### ISSUE TITLE
SuspendFunWithFlowReturnType introduced in 1.14.0 does not seem to work when activated in project

### ISSUE THREAD

**Issue by mhernand40** [detekt/detekt#3116]
## Expected Behavior
When activating the new `SuspendFunWithFlowReturnType` rule in the Detekt 1.14.0, code that consists of `suspend fun` that also returns a `Flow` should be considered a violation.

## Observed Behavior
When activating the new `SuspendFunWithFlowReturnType` rule in the Detekt 1.14.0, code that consists of `suspend fun` that also returns a `Flow` is NOT considered as a violation.

## Steps to Reproduce
1. Clone the repo https://github.com/mhernand40/DetektSuspendFunWithFlowReturnTypeTest
2. Run `./gradlew detekt`

**Expected:**
Detekt should flag a `SuspendFunWithFlowReturnType` warning for [SuspendFunWithFlowReturnTypeViolation.kt](https://github.com/mhernand40/DetektSuspendFunWithFlowReturnTypeTest/blob/master/src/main/kotlin/SuspendFunWithFlowReturnTypeViolation.kt)

**Actual:** Detekt "succeeds" (false positive).

## Context
The new `SuspendFunWithFlowReturnType` rule introduced in Detekt 1.14.0 does not seem to be working as expected when activated in a project.

## Your Environment
* Version of detekt used: 1.14.0
* Version of Gradle used (if applicable):
```
------------------------------------------------------------
Gradle 6.3
------------------------------------------------------------

Build time:   2020-03-24 19:52:07 UTC
Revision:     bacd40b727b0130eeac8855ae3f9fd9a0b207c60

Kotlin:       1.3.70
Groovy:       2.5.10
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          15 (Oracle Corporation 15+36-1562)
OS:           Mac OS X 10.15.6 x86_64
```
* Operating System and version: Mac OS X 10.15.6 x86_64
* Link to your project (if it's a public repository): https://github.com/mhernand40/DetektSuspendFunWithFlowReturnTypeTest

**Comment by mhernand40** [comment]
The [SuspendFunWithFlowReturnType](https://github.com/detekt/detekt/blob/master/detekt-rules-coroutines/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/coroutines/SuspendFunWithFlowReturnType.kt) rule attempts to resolve `Flow` types but in order to do so, the `kotlinx-coroutines-core` library needs to be in the `classpath`. The tests in [SuspendFunWithFlowReturnTypeSpec](https://github.com/detekt/detekt/blob/master/detekt-rules-coroutines/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/coroutines/SuspendFunWithFlowReturnTypeSpec.kt) run as expected because in https://github.com/detekt/detekt/pull/3098, Coroutines were added to the `KtTestCompiler` `classpath`. Coroutines was also added as a dependency within [detekt-bom/build.gradle.kts](https://github.com/detekt/detekt/blob/master/detekt-bom/build.gradle.kts#L24).

Was something else missed? Was `detekt-api` or `detekt-core` supposed to have an explicit dependency on `kotlinx-coroutines-core` as well?

**Comment by cortinico** [comment]
Given that `SuspendFunWithFlowReturnType` is using Type Resolution, you should invoke `detektMain` instead otherwise the rule will be skipped.

**Comment by mhernand40** [comment]
TIL that `detektMain` exists 🙂 

Thanks for clarifying @cortinico. I have confirmed in my sample project that if I run `./gradlew detektMain` the warning is raised. Closing this.
